src/xrpld/overlay/detail/OverlayImpl.cpp

```diff
@@ -1199,19 +1200,31 @@ OverlayImpl::getManifestsMessage()
 void
 OverlayImpl::relay(
     uint256 const& hash,
-    protocol::TMTransaction& m,
+    std::optional<std::reference_wrapper<protocol::TMTransaction>> tx,
     std::set<Peer::id_t> const& toSkip)
 {
-    auto const sm = std::make_shared<Message>(m, protocol::mtTRANSACTION);
+    std::shared_ptr<Message> sm = nullptr;
+    bool relay = tx.has_value();
+    if (relay)
+    {
+        auto& txn = tx->get();
+        sm = std::make_shared<Message>(txn, protocol::mtTRANSACTION);
+        SerialIter sit(makeSlice(txn.rawtransaction()));
+        relay = !isPseudoTx(STTx{sit});
+    }
```

Comment:
You can save the `Message` allocation if you check for `isPseudoTx` first.
```suggestion
    if (relay)
    {
        auto& txn = tx->get();
        SerialIter sit(makeSlice(txn.rawtransaction()));
        relay = !isPseudoTx(STTx{sit});
    }
    if (relay)
    {
        auto& txn = tx->get();
        sm = std::make_shared<Message>(txn, protocol::mtTRANSACTION);
    }
```